plugins/ftdidmx/FtdiWidget-libftdi.cpp

```diff
@@ -47,36 +54,169 @@ using std::vector;
 
 FtdiWidget::FtdiWidget(const string& serial,
                        const string& name,
-                       uint32_t id)
+                       uint32_t id,
+                       const int vid,
+                       const int pid
+                      )
     : m_serial(serial),
       m_name(name),
-      m_id(id) {
-  bzero(&m_handle, sizeof(struct ftdi_context));
+      m_id(id),
+      m_vid(vid),
+      m_pid(pid) {}
+
+FtdiWidget::~FtdiWidget() {
+}
+
+/**
+ * Get the number of physical interfaces our widgit has to offer.
+ *
+ * This does not deal with product names being named in a different way.
+ *
+ * Originally I had hoped to use ftdi_context::type however, it only gets set
+ * properly after the device has been opened.
+ */
+int FtdiWidget::GetInterfaceCount() {
+  std::string tmp = m_name;
+  ToLower(&tmp);
+  if (std::string::npos != tmp.find("plus4")) {
+    return 4;
+  } else  if (std::string::npos != tmp.find("plus2")) {
+    return 2;
+  } else {
+    return 1;
+  }
+}
+
+/**
+ * Build a list of all attached ftdi devices
+ */
+void FtdiWidget::Widgets(vector<FtdiWidgetInfo> *widgets) {
+  int i = -1;
+  widgets->clear();
+  struct ftdi_context *ftdi = ftdi_new();
+  if (!ftdi) {
+    OLA_WARN << "Failed to allocate FTDI context";
+    return;
+  }
+
+  const int pids[2] = { 0x6001, 0x6011 };
```

Comment:
Okay, as long as it won't be another 5 months until the next PR :), can you still do the arraysize change below please.